client/scripts/controllers/app/login.ts

```diff
@@ -27,45 +27,44 @@ function createAccount(account: Account<any>) {
   });
 }
 
-export async function setActiveAccount(account: Account<any>) {
-  return new Promise((resolve, reject) => {
-    const chain = app.activeChainId();
-    const community = app.activeCommunityId();
-    const role = app.user.getRoleInCommunity({ account, chain, community });
-
-    if (!role || role.is_user_default) {
-      app.user.ephemerallySetActiveAccount(account);
-      if (app.user.activeAccounts.filter((a) => isSameAccount(a, account)).length === 0) {
-        app.user.setActiveAccounts(app.user.activeAccounts.concat([account]));
-      }
-      resolve();
-    } else {
-      $.post(`${app.serverUrl()}/setDefaultRole`, chain ? {
-        address: account.address,
-        author_chain: account.chain.id,
-        chain,
-        jwt: app.user.jwt,
-        auth: true,
-      } : {
-        address: account.address,
-        author_chain: account.chain.id,
-        community,
-        jwt: app.user.jwt,
-        auth: true,
-      }).then((response) => {
-        // update is_user_default
-        app.user.getAllRolesInCommunity({ chain, community })
-          .forEach((r) => { r.is_user_default = false; });
-        role.is_user_default = true;
-
-        app.user.ephemerallySetActiveAccount(account);
-        if (app.user.activeAccounts.filter((a) => isSameAccount(a, account)).length === 0) {
-          app.user.setActiveAccounts(app.user.activeAccounts.concat([account]));
-        }
-        resolve();
-      }).catch((err) => reject());
+export async function setActiveAccount(account: Account<any>): Promise<void> {
+  const chain = app.activeChainId();
+  const community = app.activeCommunityId();
+  const role = app.user.getRoleInCommunity({ account, chain, community });
+
+  if (!role || role.is_user_default) {
+    app.user.ephemerallySetActiveAccount(account);
+    if (app.user.activeAccounts.filter((a) => isSameAccount(a, account)).length === 0) {
+      app.user.setActiveAccounts(app.user.activeAccounts.concat([account]));
     }
+    return;
+  }
+
+  const response = await $.post(`${app.serverUrl()}/setDefaultRole`, chain ? {
+    address: account.address,
+    author_chain: account.chain.id,
+    chain,
+    jwt: app.user.jwt,
+    auth: true,
+  } : {
+    address: account.address,
+    author_chain: account.chain.id,
+    community,
+    jwt: app.user.jwt,
+    auth: true,
   });
+  if (response.status !== 'Success') {
+    throw Error(`Unsuccessful status: ${response.status}`);
+  }
```

Comment:
If the Ajax request fails, instead of letting the exception propagate, we should carry on here (maybe displaying an error message) but avoid interrupting the flow of what is happening here, because if we throw an exception here it can propagate in some pretty unexpected ways.